controllers/integration/integration_adapter.go

```diff
@@ -137,6 +137,133 @@ func (a *Adapter) getAllApplicationSnapshots() (*[]appstudioshared.ApplicationSn
 	return &applicationSnapshots.Items, nil
 }
 
+func (a *Adapter) getAllEnvironments() (*[]appstudioshared.Environment, error) {
+
+	environmentList := &appstudioshared.EnvironmentList{}
+	opts := []client.ListOption{
+		client.InNamespace(a.application.Namespace),
+	}
+	err := a.client.List(a.context, environmentList, opts...)
+	return &environmentList.Items, err
+}
+
+func (a *Adapter) findAvailabeEnvironments() (*[]appstudioshared.Environment, error) {
```

Comment:
nit: findAvailabeEnvironments -> findAvailableEnvironments

Reply:
Do you mean the function is useless any more? Can you please make it more clear?

Reply:
And I am also wondering when & how to set ParentEnvironment for an environment? Do we need to set the ParentEnvironment after the binding is created?

Reply:
Sorry, I meant that it's a typo in the function name:
```suggestion
func (a *Adapter) findAvailableEnvironments() (*[]appstudioshared.Environment, error) {
```
We are not supposed to set the `parentEnvironment` at any point, that is set by the AppStudio users/admins themselves.

Reply:
OK. Thanks